### PR TITLE
[Core] STL Reader - Adding new entities to geometry list

### PR DESCRIPTION
**Pull request by ddiezrod**
As suggested by @philbucher in https://github.com/KratosMultiphysics/Kratos/pull/10931 I am adding new entities to the geometry container instead of the element one.

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]

File: kratos/tests/cpp_tests/input_output/test_stl_io.cpp [hunk 1]
```diff
@@ -128,13 +128,13 @@ KRATOS_TEST_CASE_IN_SUITE(WriteTriangleToSTL, KratosCoreFastSuite)
     KRATOS_CHECK_EQUAL(r_output_model_part.NumberOfProperties(), 1);
     KRATOS_CHECK_EQUAL(r_output_model_part.NumberOfSubModelParts() ,1);
     KRATOS_CHECK_EQUAL(r_output_model_part.NumberOfNodes(), 3);
-    KRATOS_CHECK_EQUAL(r_output_model_part.NumberOfElements(), 1);
+    KRATOS_CHECK_EQUAL(r_output_model_part.NumberOfGeometries(), 1);
     KRATOS_CHECK_EQUAL(r_output_model_part.NumberOfConditions(), 0);
 
     KRATOS_CHECK_EQUAL(r_output_model_part.GetSubModelPart("Main").NumberOfNodes(), 3);
-    KRATOS_CHECK_EQUAL(r_output_model_part.GetSubModelPart("Main").NumberOfElements(), 1);
+    KRATOS_CHECK_EQUAL(r_output_model_part.GetSubModelPart("Main").NumberOfGeometries(), 1);
```
Comment (philbucher) [inline kratos/tests/cpp_tests/input_output/test_stl_io.cpp:135]:
```suggestion
    KRATOS_CHECK_EQUAL(r_output_model_part.GetSubModelPart("Main").NumberOfGeometries(), 1);
    KRATOS_CHECK_EQUAL(r_output_model_part.GetSubModelPart("Main").NumberOfElements(), 0);
```

same for the other places

File: kratos/input_output/stl_io.cpp [hunk 2]
```diff
@@ -211,16 +211,13 @@ void StlIO::ReadLoop(ModelPart & rThisModelPart)
     std::size_t node_id = block_for_each<MaxReduction<std::size_t>>(
         rThisModelPart.GetRootModelPart().Nodes(),
         [](NodeType& rNode) { return rNode.Id();}) + 1;
-    std::size_t element_id = block_for_each<MaxReduction<std::size_t>>(
-        rThisModelPart.GetRootModelPart().Elements(),
-        [](Element& rElement) { return rElement.Id();}) + 1;
     Element::NodesArrayType temp_element_nodes;
```
Comment (philbucher) [inline kratos/input_output/stl_io.cpp]: this could be renamed too

**Review by philbucher — COMMENTED** [review]
minor, otherwise great

Finally we are moving towards reading geometries, and keeping the Kratos internals (aka Elements and conditions) out of the IO

This is BTW how Code_Aster does, the reading of the mesh is completely separate from the creation of the physics/finite elements

File: kratos/input_output/stl_io.cpp (hunk 2 above)
Comment (ddiezrod) [reply kratos/input_output/stl_io.cpp]: done!

**Review by philbucher — APPROVED** [review]

**Pull request merged**